src/main/java/org/fenixedu/academic/ui/struts/action/administrativeOffice/lists/StudentListByDegreeDA.java

```diff
@@ -197,17 +197,8 @@ public ActionForward searchByDegree(ActionMapping mapping, ActionForm actionForm
                 continue;
             }
 
-            if (searchBean.isConcludedInChosenYear()) {
-                Stream<ProgramConclusion> conclusions =
-                        ProgramConclusion.conclusionsFor(registration).filter(ProgramConclusion::isTerminal);
-
-                if (conclusions.allMatch(programConclusion -> {
-                    RegistrationConclusionBean conclusionBean = new RegistrationConclusionBean(registration, programConclusion);
-                    return conclusionBean.getCurriculumGroup() == null || !conclusionBean.isConcluded()
-                            || conclusionBean.getConclusionYear() != executionYear;
-                })) {
-                    continue;
-                }
+            if ((searchBean.hasAnyProgramConclusion()) && !searchBean.getProgramConclusions().contains(registration.getLastCurricularPlanProgramConclusion())) {
```

Comment:
You can use ```ProgramConclusion.conclusionsFor``` and check if any element in the returned stream matches any of the selected ```ProgramConclusion```.